expertino/clips/executor.clp

```diff
@@ -0,0 +1,74 @@
+; Copyright (c) 2024 Carologistics
+;
+; Licensed under the Apache License, Version 2.0 (the "License");
+; you may not use this file except in compliance with the License.
+; You may obtain a copy of the License at
+;
+;     http://www.apache.org/licenses/LICENSE-2.0
+;
+; Unless required by applicable law or agreed to in writing, software
+; distributed under the License is distributed on an "AS IS" BASIS,
+; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+; See the License for the specific language governing permissions and
+; limitations under the License.
+
+(defrule executor-create
+  ?pa <- (pddl-action (id ?action-id))
+  ?aa <- (agenda-action-item (action ?action-id) (execution-state SELECTED))
+  =>
+  (assert (executor (id (sym-cat EXECUTOR-(gensym*))) (pddl-action-id ?action-id) (state INIT)))
```

Comment:
should probably adapt the execution-state in the agenda-action-item. might need to add a state for that?